version: '3.8'

services:
  db:
    image: postgres:16-alpine
    restart: always
    environment:
      POSTGRES_DB: denemedb
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: Ahmetmert1
    ports:
      - "5432:5432"
    volumes:
      - db_data:/var/lib/postgresql/data
    healthcheck:
      # Düzeltme: pg_isready komutunda doğru kullanıcı adı ve veritabanı adı kullanıldı
      test: ["CMD-SHELL", "pg_isready -U postgres -d denemedb"]
      interval: 5s
      timeout: 5s
      retries: 5
      start_period: 30s

  web:
    container_name: my_fastapi_app
    build: .
    ports:
      - "8000:8000"
    environment:
      CONFIG_PATH: "/app/config.toml"
      DATABASE_HOST: db
      DATABASE_NAME: denemedb
      DATABASE_USER: postgres
      DATABASE_PASSWORD: Ahmetmert1
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - .:/app
    command: uvicorn myproject.app.myapi:app --host 0.0.0.0 --port 8000 --reload

volumes:
  db_data:
